pkg/parser/custom/terraform/terraform.go

```diff
@@ -0,0 +1,54 @@
+package terraform
+
+import (
+	"fmt"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/safedep/vet/pkg/models"
+	"os"
+
+	"github.com/hashicorp/hcl/v2/hclparse"
+)
+
+func ParseTerraformLockfile(path string) ([]lockfile.PackageDetails, error) {
```

Comment:
If you are moving this to a `graph` parser model, then the function signature should be slightly different. The `graph` parser model is more extensible i.e. we can provide config params to the parser from command line args if required in future. Unless the parser is "really" experimental and unreliable, I would suggest that we implement all parser following the `graph` parser model.

See: https://github.com/safedep/vet/blob/main/pkg/parser/npm_graph.go#L34